src/python/pants/core/goals/style.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from abc import ABCMeta
+from dataclasses import dataclass
+from typing import ClassVar, Generic, Iterable, Optional, Type, TypeVar
+
+from pants.engine.collection import Collection
+from pants.engine.fs import Snapshot
+from pants.engine.target import FieldSetWithOrigin
+from pants.util.meta import frozen_after_init
+
+_FS = TypeVar("_FS", bound=FieldSetWithOrigin)
+
+
+@frozen_after_init
+@dataclass(unsafe_hash=True)
+class StyleRequest(Generic[_FS], metaclass=ABCMeta):
```

Comment:
Maybe explicitly mention "autoformatters and linters"? This name makes me think it's only for formatters.